Tools/TestWebKitAPI/Tests/WebKitCocoa/ScreenTime.mm

```diff
@@ -546,6 +550,42 @@ - (void)userContentController:(WKUserContentController *)userContentController d
 
 #endif
 
+TEST(ScreenTime, FetchData)
+{
+    if (![PAL::getSTWebHistoryClass() instancesRespondToSelector:@selector(fetchAllHistoryWithCompletionHandler:)])
+        return;
+
+    __block bool calledScreenTimeFetchAllHistory = false;
+    InstanceMethodSwizzler swizzler {
+        PAL::getSTWebHistoryClass(),
+        @selector(fetchAllHistoryWithCompletionHandler:),
+        imp_implementationWithBlock(^(id object, NSSet<NSURL *> *urls, NSError *error) {
+            calledScreenTimeFetchAllHistory = true;
+        })
+    };
+
+    RetainPtr dataTypeScreenTime = adoptNS([[NSSet alloc] initWithArray:@[ _WKWebsiteDataTypeScreenTime ]]);
+
+    RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+
+    RetainPtr websiteDataStore = [WKWebsiteDataStore defaultDataStore];
+    [configuration setWebsiteDataStore:websiteDataStore.get()];
+
+    RetainPtr webView = webViewForScreenTimeTests(configuration.get());
+    [webView synchronouslyLoadHTMLString:@""];
+
+    __block bool done = false;
+    [websiteDataStore fetchDataRecordsOfTypes:dataTypeScreenTime.get() completionHandler:^(NSArray<WKWebsiteDataRecord *> *dataRecords) {
+        done = true;
+    }];
+
+    TestWebKitAPI::Util::run(&done);
```

Comment:
You can get rid of this, and the `done` variable, now that you have `runFor` below.

Reply:
Actually, this isn't entirely true, we still need to verify the completion handler gets called – not just that the swizzled method gets called.